test/e2e/k8s_util.go

```diff
@@ -81,25 +82,40 @@ func (k *KubernetesUtils) GetPods(ns string, key string, val string) ([]v1.Pod,
 
 // Probe execs into a Pod and checks its connectivity to another Pod.  Of course it assumes
 // that the target Pod is serving on the input port, and also that ncat is installed.
-func (k *KubernetesUtils) Probe(ns1, pod1, ns2, pod2 string, port int32) (bool, error) {
-	fromPods, err := k.GetPods(ns1, "pod", pod1)
-	if err != nil {
-		return false, errors.WithMessagef(err, "unable to get pods from ns %s", ns1)
-	}
-	if len(fromPods) == 0 {
-		return false, errors.New(fmt.Sprintf("no pod of name %s in namespace %s found", pod1, ns1))
-	}
-	fromPod := fromPods[0]
+func (k *KubernetesUtils) Probe(ns1, pod1, ns2, pod2 string, port int32, usePodName bool) (bool, error) {
+	var fromPod v1.Pod
+	var toPod v1.Pod
+
+	if usePodName {
+		pod1Returned, err := k.clientset.CoreV1().Pods(ns1).Get(context.TODO(), pod1, metav1.GetOptions{})
+		if err != nil || pod1Returned == nil {
+			return false, errors.WithMessagef(err, "unable to get Pod %s/%s", ns1, pod1)
+		}
+		fromPod = *pod1Returned
+		pod2Returned, err := k.clientset.CoreV1().Pods(ns2).Get(context.TODO(), pod2, metav1.GetOptions{})
+		if err != nil || pod2Returned == nil {
+			return false, errors.WithMessagef(err, "unable to get Pod %s/%s", ns2, pod2)
+		}
+		toPod = *pod2Returned
+	} else {
+		fromPods, err := k.GetPods(ns1, "pod", pod1)
+		if err != nil {
+			return false, errors.WithMessagef(err, "unable to get pods from ns %s", ns1)
+		}
+		if len(fromPods) == 0 {
+			return false, errors.New(fmt.Sprintf("no Pod of label pod=%s in namespace %s found", pod1, ns1))
```

Comment:
you can use `fmt.Errorf` or `error.Errorf`

Reply:
Sure.